v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-model.ts

```diff
@@ -74,10 +81,18 @@ export const UnivariateMeasureAdornmentModel = AdornmentModel
       const attrId = xAttrId ? xAttrId : yAttrId
       for (let i = 0; i < totalCount; ++i) {
         const cellKey = self.setCellKey(options, i)
-        const instanceKey = self.instanceKey(cellKey)
+        // If there are no cases in the cell, do not add an adornment
+        if (dataConfig.subPlotCases(cellKey).length === 0) continue
+
+        const instanceKey = self.instanceKey(cellKey) 
         const value = Number(self.getMeasureValue(attrId, cellKey, dataConfig))
+        // If the value is not a number, do not add an adornment
+        if (!Number.isFinite(value)) continue
```

Comment:
I suspect it will work better to go ahead and add the measure with a value of `NaN` and then deal with that possibility at render time (e.g. by not rendering). Otherwise, you have to listen for every possible circumstance that can change the value and be prepared to add/remove measures dynamically as the value changes. For instance, imagine if there were an animating slider controlling a filter that affected the number of cases -- then potentially you'd have to add/remove measures with each tick of the slider. If you always add the measures, however, then you just need to change the (volatile) values as the slider animates. This will also have the effect of preserving user-positioned labels as values appear/disappear.